clang/lib/Sema/SemaTemplateDeduction.cpp

```diff
@@ -857,7 +857,8 @@ class PackDeductionScope {
       if (auto *NTTP = dyn_cast<NonTypeTemplateParmDecl>(
               TemplateParams->getParam(Index))) {
         if (!NTTP->isExpandedParameterPack())
-          if (auto *Expansion = dyn_cast<PackExpansionType>(NTTP->getType()))
+          if (auto *Expansion = dyn_cast<PackExpansionType>(
+                  S.Context.getUnconstrainedType(NTTP->getType())))
```

Comment:
@cor3ntin I assume we are unable to deduce template arguments in the type constraint, right? This logic would otherwise end up matching the NTTP argument against the NTTP constraint

Reply:
Unclear. @zygoloid asked that exact question last year (https://lists.isocpp.org/core/2023/03/14099.php) but there are no replies yet

Reply:
Sigh. Looks like having run into another dark corner of C++ :/